kimchi/src/tests/framework.rs

```diff
@@ -52,37 +52,31 @@ pub(crate) struct TestFramework {
 pub(crate) struct TestRunner(TestFramework);
 
 impl TestFramework {
-    #[must_use]
     pub(crate) fn gates(mut self, gates: Vec<CircuitGate<Fp>>) -> Self {
         self.gates = Some(gates);
         self
     }
 
-    #[must_use]
     pub(crate) fn witness(mut self, witness: [Vec<Fp>; COLUMNS]) -> Self {
         self.witness = Some(witness);
         self
     }
 
-    #[must_use]
     pub(crate) fn public_inputs(mut self, public_inputs: Vec<Fp>) -> Self {
```

Comment:
actually, now I realize that I don't think the compiler will yell at you if there isn't a `#[must_use]`, I think the compiler yells at you when you don't use a type `Result`. In this API I think it makes sense to leave the #[must_use]

Reply:
Done https://github.com/o1-labs/proof-systems/pull/752/commits/e74530b7cf67a9cf66787be09ff530dff090630e